test/src/com/xilinx/rapidwright/edif/TestEDIFNetlist.java

```diff
@@ -287,4 +288,33 @@ public void testMigrateCellsAndSubCells(boolean uniquify) {
             Assertions.assertEquals(dstLibrary.getCells().size(), 1);
         }
     }
+
+    @Test
+    public void testBussedPortNamingCollision(@TempDir Path path) {
+        final EDIFNetlist origNetlist = EDIFTools.createNewNetlist("test");
+
+        EDIFCell top = origNetlist.getTopCell();
+
+        EDIFCellInst ff = top.createChildCellInst("ff", Design.getPrimitivesLibrary().getCell("FDRE"));
+        origNetlist.getHDIPrimitivesLibrary().addCell(ff.getCellType());
+
+        // Create two ports, one single-bit and another bussed with the same root name
+        EDIFPort port0 = top.createPort("unfortunate_name", EDIFDirection.INOUT, 1);
+        EDIFPort port1 = top.createPort("unfortunate_name[1:0]", EDIFDirection.INOUT, 2);
+
+        EDIFNet net0 = top.createNet("net0");
+        net0.createPortInst(port0);
+        net0.createPortInst("D", ff);
+
+        EDIFNet net1 = top.createNet("net1");
+        net1.createPortInst(port1, 1);
+        net1.createPortInst("R", ff);
+
+        Path tempFile = path.resolve("test.edf");
+        origNetlist.exportEDIF(tempFile);
+
+        EDIFNetlist testNetlist = EDIFTools.readEdifFile(tempFile);
+        EDIFNetlistComparator comparer = new EDIFNetlistComparator();
+        Assertions.assertEquals(0, comparer.compareNetlists(origNetlist, testNetlist));
```

Comment:
This relies on the `EDIFNetlistComparator` functioning perfectly now and into the future.

Since you've already read `testNetlist` from disk, is it possible to also do a quick check to see if both ports continue to exist, and whether they are attached to the right net?

Reply:
Added an explicit check.